pyccel/codegen/printing/ccode.py

```diff
@@ -306,24 +335,43 @@ def copy_NumpyArray_Data(self, expr):
         lhs = expr.lhs
         if rhs.rank == 0:
             raise NotImplementedError(str(expr))
+        order = lhs.order
+        transpose_arg = None
         dummy_array_name = self.scope.get_new_name('array_dummy')
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
         dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
         arg = rhs.arg if isinstance(rhs, NumpyArray) else rhs
+        if isinstance(arg, PythonList) and not isinstance(arg[0], Variable) and order=="F":
```

Comment:
In the line above it, ```arg``` is always a ```pythonList```, but from your question am guessing there will be times where it's going to be a ```PythonTuple```?

Reply:
In pyccel we don't officially support lists. We just treat them as tuples (you can see this in the definition of `PythonList` which doesn't do much except inherit from `PythonTuple`). We have some minimal support because things like:
```python
x = np.array([[1,2],[3,4]])
```
are ubiquitous.

But the expression above could just as well be written:
```python
x = np.array(((1,2),(3,4)))
```
if you respect pyccel's documentation perfectly